packages/protocol/contracts/common/EssentialContract.sol

```diff
@@ -49,6 +50,17 @@ abstract contract EssentialContract is
         _;
     }
 
+    modifier onlyFromEOA() {
+        if (msg.sender != tx.origin) revert CALLER_NOT_EOA();
+        uint256 size;
+        address sender = msg.sender;
+        assembly {
+            size := extcodesize(sender)
+        }
+        if (size != 0) revert CALLER_NOT_EOA();
+        _;
+    }
```

Comment:
I think first line (`if`) is enough as it seems to be a super set of the below code to me.

See this: https://github.com/taikoxyz/taiko-mono/pull/15100/files#r1378522913